zrml/styx/src/lib.rs

```diff
@@ -0,0 +1,116 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::too_many_arguments)]
+
+extern crate alloc;
+
+mod benchmarks;
+mod mock;
+mod tests;
+pub mod weights;
+/// Edit this file to define custom logic or remove it if it is not needed.
+/// Learn more about FRAME and the core library of Substrate FRAME pallets:
+/// <https://docs.substrate.io/v3/runtime/frame>
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+    use frame_support::{
+        pallet_prelude::*,
+        traits::{Currency, NamedReservableCurrency},
+    };
+    use frame_system::pallet_prelude::*;
+    use sp_runtime::SaturatedConversion;
+    use zeitgeist_primitives::types::Balance;
+
+    use crate::weights::WeightInfoZeitgeist;
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        /// The origin that is allowed to destroy markets.
+        type SetBurnAmountOrigin: EnsureOrigin<Self::Origin>;
+
+        /// Pallet event type
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+        /// Balances pallet
+        type Currency: NamedReservableCurrency<Self::AccountId, ReserveIdentifier = [u8; 8]>;
+
+        /// Weights generated by benchmarks
+        type WeightInfo: WeightInfoZeitgeist;
+    }
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    pub struct Pallet<T>(_);
+
+    /// Keep track of crossings. Accounts are only able to cross once.
+    #[pallet::storage]
+    pub type Crossings<T: Config> = StorageMap<_, Blake2_128Concat, T::AccountId, bool>;
```

Comment:
Since you're never using the `bool`, just replace it with the null type `()`:

```rust
pub type Crossings<T: Config> = StorageMap<_, Blake2_128Concat, T::AccountId, ()>;
```